### PR TITLE
[web] Implement setPreferredOrientation

### PR DESCRIPTION
**Pull request by ferhatb**
Related issue: https://github.com/flutter/flutter/issues/55892

### CODE REVIEW

File: lib/web_ui/lib/src/engine/dom_renderer.dart [hunk 1]
```diff
@@ -503,6 +503,74 @@ flt-glass-pane * {
...
+  static const String orientationLockTypeLandscape = 'landscape';
+  static const String orientationLockTypePortrait = 'portrait';
+  static const String orientationLockTypePortraitPrimary = 'portrait-primary';
+  static const String orientationLockTypePortraitSecondary = 'portrait-secondary';
+  static const String orientationLockTypeLandscapePrimary = 'landscape-primary';
+  static const String orientationLockTypeLandscapeSecondary = 'landscape-secondary';
+
+  /// Sets preferred screen orientation.
+  ///
+  ///
```
Comment (hterkelsen) [inline lib/web_ui/lib/src/engine/dom_renderer.dart:517]: nit: unnecessary blank line

File: lib/web_ui/lib/src/engine/dom_renderer.dart [hunk 2]
```diff
@@ -503,6 +503,74 @@ flt-glass-pane * {
...
+          return Future.value(true);
+        } else {
+          String lockType = _deviceOrientationToLockType(orientations.first);
+          if (lockType != null) {
+            final Completer<bool> completer = Completer<bool>();
+            try {
+              screenOrientation.lock(lockType).then((dynamic _) {
+                completer.complete(true);
+              }).catchError((dynamic error) {
+                // On Chrome desktop NotSupported on this device error is fired.
```
Comment (hterkelsen) [inline lib/web_ui/lib/src/engine/dom_renderer.dart:543]: nit: check that the caught error is indeed the Chrome NotSupported error?

Comment (ferhatb) [reply lib/web_ui/lib/src/engine/dom_renderer.dart:543]: Updated comment. dynamic since it is W3C Draft.

**Comment by ferhatb** [comment]
merging on infra fail

**Pull request merged**